src/pybind/mgr/dashboard/frontend/src/app/shared/services/prometheus-alert.service.ts

```diff
@@ -64,18 +64,18 @@ export class PrometheusAlertService {
       this.notifyOnAlertChanges(alerts, this.alerts);
     }
     this.activeAlerts = _.reduce<AlertmanagerAlert, number>(
-      this.alerts,
+      alerts,
       (result, alert) => (alert.status.state === 'active' ? ++result : result),
       0
     );
```

Comment:
Looks like active alerts are not sorted right now, can you do that as well?

Reply:
how would you want me to sort it? Recent ones first or the `critical first and warnings later`. As a user, I think I'd prefer to have the recent ones shown to me first. Or I prefer to have a control for the filtering. Either recent or the type of alert.

Reply:
Can we sort it via the recent critical alerts?

Reply:
but that would mean, the recent warnings are lost below in the list right? Atleast in the Status card where there is no way to sort and force the user to scroll down?

Reply:
maybe i can sort it with recent alerts for now in the status card and later on create a toggle to sort critical-warning alerts?

Reply:
For recent warning , user has options to filter by the warning and have look if they want.  By clicking on warning they can have a look at recent warnings anytime. As warning holds less importance when they have critical alerts.

Reply:
if you are okay, i'd like to take this up in a different PR, preferrably something like this: https://github.com/ceph/ceph/pull/53679.

I was wondering about the sorting and the processing we have to do in the front-end. Then I saw the grouping in the prometheus alerts. So thought this could be nicer if we can do it from the backend rather than processing it in the frontend.

Reply:
Ideally best to be done from backend.